src/components/dropdown/dropdown.ts

```diff
@@ -0,0 +1,569 @@
+  ESCAPE = 'escape',
+  ENTER = 'enter',
+}
+
+export interface IgcDropDownEventMap extends IgcToggleEventMap {
+  igcChange: CustomEvent<ISelectionChangeEventArgs>;
+}
+
+export interface ISelectionChangeEventArgs {
+  newItem: IgcDropDownItemComponent | null;
+}
+
+/**
+ * Represents a DropDown component.
+ *
+ * @element igc-dropdown
+ *
+ * @fires igcChange - Emitted when the selected item changes.
+ *
+ * @slot target - Renders the dropdown's target element.
+ * @slot - Renders the dropdown list items.
+ *
+ *
+ */
+export default class IgcDropDownComponent extends EventEmitterMixin<
+  IgcDropDownEventMap,
+  Constructor<LitElement>
+>(LitElement) {
+  public static readonly tagName = 'igc-dropdown';
+
+  public static styles = styles;
+
+  private toggleController!: IgcToggleController;
+  private selectedItem!: IgcDropDownItemComponent | null;
+  private activeItem!: IgcDropDownItemComponent;
+  private target!: HTMLElement;
+
+  private get allItems(): IgcDropDownItemComponent[] {
+    const groupItems: IgcDropDownItemComponent[] = this.groups.flatMap(
+      (group) => group.items
+    );
+    return [...this.items, ...groupItems];
+  }
+
+  @query('#igcDDLContent')
+  protected content!: HTMLElement;
+
+  @query('#igcScrollContainer')
+  protected scrollContainer!: HTMLElement;
+
+  @queryAssignedElements({ flatten: true, selector: 'igc-dropdown-item' })
+  protected items!: Array<IgcDropDownItemComponent>;
+
+  @queryAssignedElements({ flatten: true, selector: 'igc-dropdown-group' })
+  protected groups!: Array<IgcDropDownGroupComponent>;
+
+  @queryAssignedElements({ slot: 'target' })
+  private targetNodes!: Array<HTMLElement>;
+
+  /** Sets the open state of the dropdown list. */
+  @property({ type: Boolean })
+  public open = false;
+
+  /** Sets the dropdown list's positioning strategy. */
+  @property()
+  public positionStrategy: 'absolute' | 'fixed' = 'absolute';
+
+  /** The preferred placement of the dropdown list around the target element.
+   * @type {"top" | "top-start" | "top-end" | "bottom" | "bottom-start" | "bottom-end" | "right" | "right-start" | "right-end" | "left" | "left-start" | "left-end"}
+   */
+  @property()
+  public placement: IgcPlacement = 'bottom-start';
+
+  /**
+   * Whether the list should be flipped to the opposite side of the target once it's about to overflow the visible area.
+   * When true, once enough space is detected on its preferred side, it will flip back.
+   */
+  @property({ type: Boolean })
+  public flip = false;
+
+  /** Whether the dropdown should be hidden on clicking outside of it. */
+  @property({ type: Boolean })
+  public closeOnOutsideClick = true;
+
+  /** Determines the behavior of the dropdown list during scrolling the container. */
+  @property()
+  public scrollStrategy: 'scroll' | 'close' | 'block' | 'noop' = 'noop';
+
+  /** The amount of offset in horizontal and/or vertical direction. */
+  @property()
+  public offset!: { x: number; y: number };
```

Comment:
We could possibly implement custom attribute converter for this one that uses comma-separated string.